pypeit/spectrographs/mdm_osmos.py

```diff
@@ -233,3 +234,135 @@ def check_frame_type(self, ftype, fitstbl, exprng=None):
         msgs.warn('Cannot determine if frames are of type {0}.'.format(ftype))
         return np.zeros(len(fitstbl), dtype=bool)
 
+
+class MDMOSMOSR4KSpectrograph(MDMOSMOSMDM4KSpectrograph):
+    """
+    Child to handle MDM OSMOS R4K instrument+detector
+    """
+    ndet = 1
+    name = 'mdm_osmos_r4k'
+    camera = 'R4K'
+    url = 'https://www.astronomy.ohio-state.edu/martini.10/osmos/'
+    header_name = 'OSMOS'
+    supported = True
+    comment = 'MDM OSMOS spectrometer for the red. Requires calibratoins windowed down to the science frame.'
+
+    def get_detector_par(self, det, hdu=None):
+        """
+        Return metadata for the selected detector.
+
+        THIS IS FOR WINDOWED SCIENCE FRAMES
+        AND WE ARE HACKING THE CALIBS
```

Comment:
We should consider providing a script that does this.  I realize this is pretty specific because it's only needed for this instrument, but I imagine users would be happier calling something like `pypeit_prep_data mdm_osmos_4k *.fits` then having to dig up the notebook.  That script can apply to more instruments as needed (e.g., we could use it to help people add the slit info from tilsotua for LRIS).